File: day_01.py
x.split("\n")
x.split("\n\n")
elves = x.split("\n\n")
values = [sum([int(a) for a in elf.split("\n")]) for elf in elves]

# Task 1
max(values)

values.sort(reverse=True)
values[:3]

# Task 2
sum(values[:3])

File: day_01.rs
fn main() {
    let raw_input = std::fs::read_to_string("./input_01.txt").expect("failed to read input");
    let input: Vec<&str> = raw_input.lines().collect();
    let mut elves: Vec<i64> = input.split(|s| s.is_empty())
                .map(|elf| elf.iter().map(|s| s.parse::<i64>().expect("no snack"))
                                .sum())
                .collect();
    let task1 = elves.iter().max().expect("Couldn't find max");
    println!("Task 1: {}", task1);
    elves.sort();
    let top3 = elves.as_slice()[elves.len() - 3..].to_vec();
    let task2: i64 = top3.iter().sum();
    println!("Task 2: {}", task2);
}
